[*] Start mutation process:
   - targets: graph2.py
   - tests: ts-3-5/test_3-5_0-9_274.py
[*] 7 tests passed:
   - test_3-5_0-9_274 [0.19778 s]
[*] Start mutants generation and execution:
   - [#   1] CDI graph2: 
--------------------------------------------------------------------------------
  22:     adjacent(n1, n2): returns True if there is an edge connecting n1 and n2,
  23:     False if not, raises an error if either of the supplied nodes are not in g.
  24:     """
  25:     
- 26:     def __init__(self, data=None):
- 27:         '''Initialize graph.'''
- 28:         self.graph = {}
- 29:         if data:
- 30:             for i in data:
- 31:                 self.add_node(i)
- 32:     
+ 26:     @classmethod
+ 27:     def __init__(self, data=None):
+ 28:         '''Initialize graph.'''
+ 29:         self.graph = {}
+ 30:         if data:
+ 31:             for i in data:
+ 32:                 self.add_node(i)
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.19177 s] survived
   - [#   2] CDI graph2: 
--------------------------------------------------------------------------------
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
- 33:     def nodes(self):
- 34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
- 36:     
+ 33:     @classmethod
+ 34:     def nodes(self):
+ 35:         '''Return a list of all nodes in the graph.'''
+ 36:         return list(self.graph.keys())
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
--------------------------------------------------------------------------------
[0.21864 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_node

empty_graph = <graph2.Graph object at 0x7c01b3717d90>

    def test_add_node(empty_graph):
        empty_graph.add_node(1)
>       assert 1 in empty_graph.nodes()

ts-3-5/test_3-5_0-9_274.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:36: AttributeError
   - [#   3] CDI graph2: 
--------------------------------------------------------------------------------
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
- 37:     def edges(self):
- 38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
- 40:     
+ 37:     @classmethod
+ 38:     def edges(self):
+ 39:         '''Return a list of all edges in the graph.'''
+ 40:         return [edge for edges in self.graph.values() for edge in edges]
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
--------------------------------------------------------------------------------
[0.18972 s] survived
   - [#   4] CDI graph2: 
--------------------------------------------------------------------------------
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
- 41:     def add_node(self, n):
- 42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
- 44:     
+ 41:     @classmethod
+ 42:     def add_node(self, n):
+ 43:         '''Add a new node to the graph.'''
+ 44:         self.graph.setdefault(n, set())
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
--------------------------------------------------------------------------------
[0.22026 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_node

empty_graph = <graph2.Graph object at 0x7c01b3768ca0>

    def test_add_node(empty_graph):
>       empty_graph.add_node(1)

ts-3-5/test_3-5_0-9_274.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 1

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:44: AttributeError
   - [#   5] CDI graph2: 
--------------------------------------------------------------------------------
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
- 45:     def add_edge(self, n1, n2):
- 46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
- 50:     
+ 45:     @classmethod
+ 46:     def add_edge(self, n1, n2):
+ 47:         '''Add new edge to the graph.'''
+ 48:         self.graph.setdefault(n1, set([n2]))
+ 49:         self.graph.setdefault(n2, set())
+ 50:         self.graph[n1].add(n2)
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
--------------------------------------------------------------------------------
[0.21037 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b34882e0>

    def test_add_edge(empty_graph):
>       empty_graph.add_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n1 = 1, n2 = 2

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:48: AttributeError
   - [#   6] CDI graph2: 
--------------------------------------------------------------------------------
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
- 51:     def del_node(self, n):
- 52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
- 54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
- 56:     
+ 51:     @classmethod
+ 52:     def del_node(self, n):
+ 53:         """Delete the node 'n' from the graph."""
+ 54:         del self.graph[n]
+ 55:         for k in self.graph:
+ 56:             self.graph[k].discard(n)
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.20979 s] killed by ts-3-5/test_3-5_0-9_274.py::test_del_node

empty_graph = <graph2.Graph object at 0x7c01b376b6a0>

    def test_del_node(empty_graph):
        empty_graph.add_node(1)
>       empty_graph.del_node(1)

ts-3-5/test_3-5_0-9_274.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 1

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:54: AttributeError
   - [#   7] CDI graph2: 
--------------------------------------------------------------------------------
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
- 57:     def del_edge(self, n1, n2):
- 58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
- 60:     
+ 57:     @classmethod
+ 58:     def del_edge(self, n1, n2):
+ 59:         '''Delete the edge connecting n1 and n2.'''
+ 60:         self.graph[n1].remove(n2)
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
--------------------------------------------------------------------------------
[0.21940 s] killed by ts-3-5/test_3-5_0-9_274.py::test_del_edge

empty_graph = <graph2.Graph object at 0x7c01b36d4280>

    def test_del_edge(empty_graph):
        empty_graph.add_edge(1, 2)
>       empty_graph.del_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n1 = 1, n2 = 2

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:60: AttributeError
   - [#   8] CDI graph2: 
--------------------------------------------------------------------------------
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
- 61:     def has_node(self, n):
- 62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
- 64:     
+ 61:     @classmethod
+ 62:     def has_node(self, n):
+ 63:         """Return boolean if 'n' is in the graph."""
+ 64:         return n in self.graph
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
--------------------------------------------------------------------------------
[0.20767 s] killed by ts-3-5/test_3-5_0-9_274.py::test_has_node

empty_graph = <graph2.Graph object at 0x7c01b3714a00>

    def test_has_node(empty_graph):
        empty_graph.add_node(1)
>       assert empty_graph.has_node(1) == True

ts-3-5/test_3-5_0-9_274.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 1

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:64: AttributeError
   - [#   9] CDI graph2: 
--------------------------------------------------------------------------------
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
- 65:     def neighbors(self, n):
- 66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
- 68:     
+ 65:     @classmethod
+ 66:     def neighbors(self, n):
+ 67:         '''Return the list of all nodes connected to n by edges.'''
+ 68:         return self.graph[n]
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21286 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b397e650>

    def test_add_edge(empty_graph):
        empty_graph.add_edge(1, 2)
>       assert 2 in empty_graph.neighbors(1)

ts-3-5/test_3-5_0-9_274.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'graph2.Graph'>, n = 1

>   ???
E   AttributeError: type object 'Graph' has no attribute 'graph'

graph2:68: AttributeError
   - [#  10] CDI graph2: 
--------------------------------------------------------------------------------
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
- 69:     def adjacent(self, n1, n2):
- 70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 69:     @classmethod
+ 70:     def adjacent(self, n1, n2):
+ 71:         '''Return boolean if there is an edge connecting n1 and n2.'''
+ 72:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21551 s] incompetent

TypeError: graph2:72: TypeError: Graph.neighbors() missing 1 required positional argument: 'n'

   - [#  11] COI graph2: 
--------------------------------------------------------------------------------
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
  28:         self.graph = {}
- 29:         if data:
+ 29:         if not data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
--------------------------------------------------------------------------------
[0.21242 s] incompetent

TypeError: graph2:30: TypeError: 'NoneType' object is not iterable

   - [#  12] COI graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         return n not in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.21809 s] killed by ts-3-5/test_3-5_0-9_274.py::test_has_node

empty_graph = <graph2.Graph object at 0x7c01b35713f0>

    def test_has_node(empty_graph):
        empty_graph.add_node(1)
>       assert empty_graph.has_node(1) == True
E       AssertionError

ts-3-5/test_3-5_0-9_274.py:35: AssertionError
   - [#  13] COI graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         return n2 not in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.22261 s] killed by ts-3-5/test_3-5_0-9_274.py::test_adjacent

empty_graph = <graph2.Graph object at 0x7c01b3765570>

    def test_adjacent(empty_graph):
        empty_graph.add_edge(1, 2)
>       assert empty_graph.adjacent(1, 2) == True
E       AssertionError

ts-3-5/test_3-5_0-9_274.py:43: AssertionError
   - [#  14] OIL graph2: 
--------------------------------------------------------------------------------
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
- 32:     
+ 32:                 break
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.17725 s] survived
   - [#  15] OIL graph2: 
--------------------------------------------------------------------------------
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
- 56:     
+ 56:             break
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.19207 s] survived
   - [#  16] RIL graph2: 
--------------------------------------------------------------------------------
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
- 30:             for i in data:
+ 30:             for i in reversed(data):
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
--------------------------------------------------------------------------------
[0.19391 s] survived
   - [#  17] RIL graph2: 
--------------------------------------------------------------------------------
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
- 54:         for k in self.graph:
+ 54:         for k in reversed(self.graph):
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
--------------------------------------------------------------------------------
[0.18927 s] survived
   - [#  18] SDI graph2: 
--------------------------------------------------------------------------------
  22:     adjacent(n1, n2): returns True if there is an edge connecting n1 and n2,
  23:     False if not, raises an error if either of the supplied nodes are not in g.
  24:     """
  25:     
- 26:     def __init__(self, data=None):
- 27:         '''Initialize graph.'''
- 28:         self.graph = {}
- 29:         if data:
- 30:             for i in data:
- 31:                 self.add_node(i)
- 32:     
+ 26:     @staticmethod
+ 27:     def __init__(self, data=None):
+ 28:         '''Initialize graph.'''
+ 29:         self.graph = {}
+ 30:         if data:
+ 31:             for i in data:
+ 32:                 self.add_node(i)
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
--------------------------------------------------------------------------------
[0.21490 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:13: TypeError: Graph.__init__() missing 1 required positional argument: 'self'

   - [#  19] SDI graph2: 
--------------------------------------------------------------------------------
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
- 33:     def nodes(self):
- 34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
- 36:     
+ 33:     @staticmethod
+ 34:     def nodes(self):
+ 35:         '''Return a list of all nodes in the graph.'''
+ 36:         return list(self.graph.keys())
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
--------------------------------------------------------------------------------
[0.22074 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:17: TypeError: Graph.nodes() missing 1 required positional argument: 'self'

   - [#  20] SDI graph2: 
--------------------------------------------------------------------------------
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
  36:     
- 37:     def edges(self):
- 38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
- 40:     
+ 37:     @staticmethod
+ 38:     def edges(self):
+ 39:         '''Return a list of all edges in the graph.'''
+ 40:         return [edge for edges in self.graph.values() for edge in edges]
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
--------------------------------------------------------------------------------
[0.17337 s] survived
   - [#  21] SDI graph2: 
--------------------------------------------------------------------------------
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
- 41:     def add_node(self, n):
- 42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
- 44:     
+ 41:     @staticmethod
+ 42:     def add_node(self, n):
+ 43:         '''Add a new node to the graph.'''
+ 44:         self.graph.setdefault(n, set())
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
--------------------------------------------------------------------------------
[0.21187 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:16: TypeError: Graph.add_node() missing 1 required positional argument: 'n'

   - [#  22] SDI graph2: 
--------------------------------------------------------------------------------
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
  44:     
- 45:     def add_edge(self, n1, n2):
- 46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
- 50:     
+ 45:     @staticmethod
+ 46:     def add_edge(self, n1, n2):
+ 47:         '''Add new edge to the graph.'''
+ 48:         self.graph.setdefault(n1, set([n2]))
+ 49:         self.graph.setdefault(n2, set())
+ 50:         self.graph[n1].add(n2)
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
--------------------------------------------------------------------------------
[0.22613 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:20: TypeError: Graph.add_edge() missing 1 required positional argument: 'n2'

   - [#  23] SDI graph2: 
--------------------------------------------------------------------------------
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
- 51:     def del_node(self, n):
- 52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
- 54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
- 56:     
+ 51:     @staticmethod
+ 52:     def del_node(self, n):
+ 53:         """Delete the node 'n' from the graph."""
+ 54:         del self.graph[n]
+ 55:         for k in self.graph:
+ 56:             self.graph[k].discard(n)
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
--------------------------------------------------------------------------------
[0.23132 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:25: TypeError: Graph.del_node() missing 1 required positional argument: 'n'

   - [#  24] SDI graph2: 
--------------------------------------------------------------------------------
  53:         del self.graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
- 57:     def del_edge(self, n1, n2):
- 58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
- 60:     
+ 57:     @staticmethod
+ 58:     def del_edge(self, n1, n2):
+ 59:         '''Delete the edge connecting n1 and n2.'''
+ 60:         self.graph[n1].remove(n2)
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
--------------------------------------------------------------------------------
[0.21207 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:30: TypeError: Graph.del_edge() missing 1 required positional argument: 'n2'

   - [#  25] SDI graph2: 
--------------------------------------------------------------------------------
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
  60:     
- 61:     def has_node(self, n):
- 62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
- 64:     
+ 61:     @staticmethod
+ 62:     def has_node(self, n):
+ 63:         """Return boolean if 'n' is in the graph."""
+ 64:         return n in self.graph
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
--------------------------------------------------------------------------------
[0.22678 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:35: TypeError: Graph.has_node() missing 1 required positional argument: 'n'

   - [#  26] SDI graph2: 
--------------------------------------------------------------------------------
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
  64:     
- 65:     def neighbors(self, n):
- 66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
- 68:     
+ 65:     @staticmethod
+ 66:     def neighbors(self, n):
+ 67:         '''Return the list of all nodes connected to n by edges.'''
+ 68:         return self.graph[n]
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.19258 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:21: TypeError: Graph.neighbors() missing 1 required positional argument: 'n'

   - [#  27] SDI graph2: 
--------------------------------------------------------------------------------
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
  68:     
- 69:     def adjacent(self, n1, n2):
- 70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 69:     @staticmethod
+ 70:     def adjacent(self, n1, n2):
+ 71:         '''Return boolean if there is an edge connecting n1 and n2.'''
+ 72:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21554 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:43: TypeError: Graph.adjacent() missing 1 required positional argument: 'n2'

   - [#  28] SDL graph2: 
--------------------------------------------------------------------------------
  24:     """
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
- 28:         self.graph = {}
+ 28:         pass
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
--------------------------------------------------------------------------------
[0.21332 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_node

empty_graph = <graph2.Graph object at 0x7c01b3769450>

    def test_add_node(empty_graph):
>       empty_graph.add_node(1)

ts-3-5/test_3-5_0-9_274.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b3769450>, n = 1

>   ???
E   AttributeError: 'Graph' object has no attribute 'graph'

graph2:43: AttributeError
   - [#  29] SDL graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 pass
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.19251 s] survived
   - [#  30] SDL graph2: 
--------------------------------------------------------------------------------
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
+ 35:         pass
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
--------------------------------------------------------------------------------
[0.21133 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:17: TypeError: argument of type 'NoneType' is not iterable

   - [#  31] SDL graph2: 
--------------------------------------------------------------------------------
  35:         return list(self.graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
+ 39:         pass
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
--------------------------------------------------------------------------------
[0.19856 s] survived
   - [#  32] SDL graph2: 
--------------------------------------------------------------------------------
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
+ 43:         pass
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
--------------------------------------------------------------------------------
[0.21073 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_node

empty_graph = <graph2.Graph object at 0x7c01b37a7250>

    def test_add_node(empty_graph):
        empty_graph.add_node(1)
>       assert 1 in empty_graph.nodes()
E       AssertionError

ts-3-5/test_3-5_0-9_274.py:17: AssertionError
   - [#  33] SDL graph2: 
--------------------------------------------------------------------------------
  43:         self.graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
+ 47:         pass
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
--------------------------------------------------------------------------------
[0.18917 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b35026e0>

    def test_add_edge(empty_graph):
>       empty_graph.add_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b35026e0>, n1 = 1, n2 = 2

>   ???
E   KeyError: 1

graph2:49: KeyError
   - [#  34] SDL graph2: 
--------------------------------------------------------------------------------
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
+ 48:         pass
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
--------------------------------------------------------------------------------
[0.19095 s] survived
   - [#  35] SDL graph2: 
--------------------------------------------------------------------------------
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
+ 49:         pass
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
--------------------------------------------------------------------------------
[0.19591 s] survived
   - [#  36] SDL graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             pass
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.19042 s] survived
   - [#  37] SDL graph2: 
--------------------------------------------------------------------------------
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
+ 59:         pass
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
--------------------------------------------------------------------------------
[0.21485 s] killed by ts-3-5/test_3-5_0-9_274.py::test_del_edge

empty_graph = <graph2.Graph object at 0x7c01b37a7c40>

    def test_del_edge(empty_graph):
        empty_graph.add_edge(1, 2)
        empty_graph.del_edge(1, 2)
>       assert 2 not in empty_graph.neighbors(1)
E       AssertionError

ts-3-5/test_3-5_0-9_274.py:31: AssertionError
   - [#  38] SDL graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         pass
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.21997 s] killed by ts-3-5/test_3-5_0-9_274.py::test_has_node

empty_graph = <graph2.Graph object at 0x7c01b353ea70>

    def test_has_node(empty_graph):
        empty_graph.add_node(1)
>       assert empty_graph.has_node(1) == True
E       AssertionError

ts-3-5/test_3-5_0-9_274.py:35: AssertionError
   - [#  39] SDL graph2: 
--------------------------------------------------------------------------------
  63:         return n in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
+ 67:         pass
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.20369 s] incompetent

TypeError: /home/lucca/teste/python_experiments2/graph2/ts-3-5/test_3-5_0-9_274.py:21: TypeError: argument of type 'NoneType' is not iterable

   - [#  40] SDL graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         pass
--------------------------------------------------------------------------------
[0.19049 s] killed by ts-3-5/test_3-5_0-9_274.py::test_adjacent

empty_graph = <graph2.Graph object at 0x7c01b37a4820>

    def test_adjacent(empty_graph):
        empty_graph.add_edge(1, 2)
>       assert empty_graph.adjacent(1, 2) == True
E       AssertionError

ts-3-5/test_3-5_0-9_274.py:43: AssertionError
   - [#  41] SVD graph2: 
--------------------------------------------------------------------------------
  24:     """
  25:     
  26:     def __init__(self, data=None):
  27:         '''Initialize graph.'''
- 28:         self.graph = {}
+ 28:         graph = {}
  29:         if data:
  30:             for i in data:
  31:                 self.add_node(i)
  32:     
--------------------------------------------------------------------------------
[0.00000 s] incompetent

ValueError: expression must have Store context but has Load instead

   - [#  42] SVD graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.19255 s] survived
   - [#  43] SVD graph2: 
--------------------------------------------------------------------------------
  31:                 self.add_node(i)
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
- 35:         return list(self.graph.keys())
+ 35:         return list(graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
  39:         return [edge for edges in self.graph.values() for edge in edges]
--------------------------------------------------------------------------------
[0.23288 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_node

empty_graph = <graph2.Graph object at 0x7c01b356cf10>

    def test_add_node(empty_graph):
        empty_graph.add_node(1)
>       assert 1 in empty_graph.nodes()

ts-3-5/test_3-5_0-9_274.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b356cf10>

>   ???
E   NameError: name 'graph' is not defined

graph2:35: NameError
   - [#  44] SVD graph2: 
--------------------------------------------------------------------------------
  35:         return list(self.graph.keys())
  36:     
  37:     def edges(self):
  38:         '''Return a list of all edges in the graph.'''
- 39:         return [edge for edges in self.graph.values() for edge in edges]
+ 39:         return [edge for edges in graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
  43:         self.graph.setdefault(n, set())
--------------------------------------------------------------------------------
[0.18977 s] survived
   - [#  45] SVD graph2: 
--------------------------------------------------------------------------------
  39:         return [edge for edges in self.graph.values() for edge in edges]
  40:     
  41:     def add_node(self, n):
  42:         '''Add a new node to the graph.'''
- 43:         self.graph.setdefault(n, set())
+ 43:         graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
--------------------------------------------------------------------------------
[0.21192 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_node

empty_graph = <graph2.Graph object at 0x7c01b37163b0>

    def test_add_node(empty_graph):
>       empty_graph.add_node(1)

ts-3-5/test_3-5_0-9_274.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b37163b0>, n = 1

>   ???
E   NameError: name 'graph' is not defined

graph2:43: NameError
   - [#  46] SVD graph2: 
--------------------------------------------------------------------------------
  43:         self.graph.setdefault(n, set())
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
- 47:         self.graph.setdefault(n1, set([n2]))
+ 47:         graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
--------------------------------------------------------------------------------
[0.23782 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b34f3400>

    def test_add_edge(empty_graph):
>       empty_graph.add_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b34f3400>, n1 = 1, n2 = 2

>   ???
E   NameError: name 'graph' is not defined

graph2:47: NameError
   - [#  47] SVD graph2: 
--------------------------------------------------------------------------------
  44:     
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
- 48:         self.graph.setdefault(n2, set())
+ 48:         graph.setdefault(n2, set())
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
--------------------------------------------------------------------------------
[0.19371 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b33ff730>

    def test_add_edge(empty_graph):
>       empty_graph.add_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b33ff730>, n1 = 1, n2 = 2

>   ???
E   NameError: name 'graph' is not defined

graph2:48: NameError
   - [#  48] SVD graph2: 
--------------------------------------------------------------------------------
  45:     def add_edge(self, n1, n2):
  46:         '''Add new edge to the graph.'''
  47:         self.graph.setdefault(n1, set([n2]))
  48:         self.graph.setdefault(n2, set())
- 49:         self.graph[n1].add(n2)
+ 49:         graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
--------------------------------------------------------------------------------
[0.21683 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b3791840>

    def test_add_edge(empty_graph):
>       empty_graph.add_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:20: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b3791840>, n1 = 1, n2 = 2

>   ???
E   NameError: name 'graph' is not defined

graph2:49: NameError
   - [#  49] SVD graph2: 
--------------------------------------------------------------------------------
  49:         self.graph[n1].add(n2)
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
- 53:         del self.graph[n]
+ 53:         del graph[n]
  54:         for k in self.graph:
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
--------------------------------------------------------------------------------
[0.21478 s] killed by ts-3-5/test_3-5_0-9_274.py::test_del_node

empty_graph = <graph2.Graph object at 0x7c01b37178e0>

    def test_del_node(empty_graph):
        empty_graph.add_node(1)
>       empty_graph.del_node(1)

ts-3-5/test_3-5_0-9_274.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b37178e0>, n = 1

>   ???
E   NameError: name 'graph' is not defined

graph2:53: NameError
   - [#  50] SVD graph2: 
--------------------------------------------------------------------------------
  50:     
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
- 54:         for k in self.graph:
+ 54:         for k in graph:
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
--------------------------------------------------------------------------------
[0.21226 s] killed by ts-3-5/test_3-5_0-9_274.py::test_del_node

empty_graph = <graph2.Graph object at 0x7c01b34f0fd0>

    def test_del_node(empty_graph):
        empty_graph.add_node(1)
>       empty_graph.del_node(1)

ts-3-5/test_3-5_0-9_274.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b34f0fd0>, n = 1

>   ???
E   NameError: name 'graph' is not defined

graph2:54: NameError
   - [#  51] SVD graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.19321 s] survived
   - [#  52] SVD graph2: 
--------------------------------------------------------------------------------
  55:             self.graph[k].discard(n)
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
- 59:         self.graph[n1].remove(n2)
+ 59:         graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
  63:         return n in self.graph
--------------------------------------------------------------------------------
[0.21748 s] killed by ts-3-5/test_3-5_0-9_274.py::test_del_edge

empty_graph = <graph2.Graph object at 0x7c01b3792dd0>

    def test_del_edge(empty_graph):
        empty_graph.add_edge(1, 2)
>       empty_graph.del_edge(1, 2)

ts-3-5/test_3-5_0-9_274.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b3792dd0>, n1 = 1, n2 = 2

>   ???
E   NameError: name 'graph' is not defined

graph2:59: NameError
   - [#  53] SVD graph2: 
--------------------------------------------------------------------------------
  59:         self.graph[n1].remove(n2)
  60:     
  61:     def has_node(self, n):
  62:         """Return boolean if 'n' is in the graph."""
- 63:         return n in self.graph
+ 63:         return n in graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
  67:         return self.graph[n]
--------------------------------------------------------------------------------
[0.20335 s] killed by ts-3-5/test_3-5_0-9_274.py::test_has_node

empty_graph = <graph2.Graph object at 0x7c01b37158d0>

    def test_has_node(empty_graph):
        empty_graph.add_node(1)
>       assert empty_graph.has_node(1) == True

ts-3-5/test_3-5_0-9_274.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b37158d0>, n = 1

>   ???
E   NameError: name 'graph' is not defined

graph2:63: NameError
   - [#  54] SVD graph2: 
--------------------------------------------------------------------------------
  63:         return n in self.graph
  64:     
  65:     def neighbors(self, n):
  66:         '''Return the list of all nodes connected to n by edges.'''
- 67:         return self.graph[n]
+ 67:         return graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
  71:         return n2 in self.neighbors(n1)
--------------------------------------------------------------------------------
[0.21414 s] killed by ts-3-5/test_3-5_0-9_274.py::test_add_edge

empty_graph = <graph2.Graph object at 0x7c01b34f0820>

    def test_add_edge(empty_graph):
        empty_graph.add_edge(1, 2)
>       assert 2 in empty_graph.neighbors(1)

ts-3-5/test_3-5_0-9_274.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b34f0820>, n = 1

>   ???
E   NameError: name 'graph' is not defined

graph2:67: NameError
   - [#  55] SVD graph2: 
--------------------------------------------------------------------------------
  67:         return self.graph[n]
  68:     
  69:     def adjacent(self, n1, n2):
  70:         '''Return boolean if there is an edge connecting n1 and n2.'''
- 71:         return n2 in self.neighbors(n1)
+ 71:         return n2 in neighbors(n1)
--------------------------------------------------------------------------------
[0.21837 s] killed by ts-3-5/test_3-5_0-9_274.py::test_adjacent

empty_graph = <graph2.Graph object at 0x7c01b3850eb0>

    def test_adjacent(empty_graph):
        empty_graph.add_edge(1, 2)
>       assert empty_graph.adjacent(1, 2) == True

ts-3-5/test_3-5_0-9_274.py:43: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <graph2.Graph object at 0x7c01b3850eb0>, n1 = 1, n2 = 2

>   ???
E   NameError: name 'neighbors' is not defined

graph2:71: NameError
   - [#  56] ZIL graph2: 
--------------------------------------------------------------------------------
  27:         '''Initialize graph.'''
  28:         self.graph = {}
  29:         if data:
  30:             for i in data:
- 31:                 self.add_node(i)
+ 31:                 break
  32:     
  33:     def nodes(self):
  34:         '''Return a list of all nodes in the graph.'''
  35:         return list(self.graph.keys())
--------------------------------------------------------------------------------
[0.18893 s] survived
   - [#  57] ZIL graph2: 
--------------------------------------------------------------------------------
  51:     def del_node(self, n):
  52:         """Delete the node 'n' from the graph."""
  53:         del self.graph[n]
  54:         for k in self.graph:
- 55:             self.graph[k].discard(n)
+ 55:             break
  56:     
  57:     def del_edge(self, n1, n2):
  58:         '''Delete the edge connecting n1 and n2.'''
  59:         self.graph[n1].remove(n2)
--------------------------------------------------------------------------------
[0.19275 s] survived
[*] Mutation score [13.52231 s]: 60.5%
   - all: 57
   - killed: 26 (45.6%)
   - survived: 17 (29.8%)
   - incompetent: 14 (24.6%)
   - timeout: 0 (0.0%)
